### ISSUE TITLE
PULSE / TIME OUT timer's not resetting on passed through joystick events

### ISSUE THREAD

**Issue by oneswitch** [evilC/OneSwitch-Utilities#10]
Not sure if it's possible, but if I use button 2 as a pass through, to speed along the scanning speed, sometimes it skips down twice. This is because there seems to be no timer reset on PULSE (and I guess not TIME OUT either).

If possible, the pass through joystick should reset the two timers, just as pressing the choice button does.

**Comment by evilC** [comment]
Yes this is possible, but would you want it to happen in all cases?

**Comment by oneswitch** [comment]
Definitely for anything that's passed through as joystick button 2 or 22 (both used in JTK as a way to move down the menu system). Also for 3 or 23 (both used as a Time Out condition). That's critical I think to avoid this issue. Think that would do the trick.

**Comment by evilC** [comment]
So it seems to me that they are not going to have 20+ buttons on their stick, so we just need to not pass through buttons 2 or 3?

**Comment by oneswitch** [comment]
It would be useful to add some extra bindable "choice" buttons for those who can manage two or three physical switches (either alone or in a team). It would also support people using a keyboard-HID switch interface. I suppose this would end up being a lot like "Mixer".

If any input is detected (choice 1 -4 for instance) then the Pulse and Timeout clock should be reset.

Does this sound do-able, or a nightmare? Especially in relation to the "special keys" tab idea, and the microphone-switches?

Barrie
